hrm-domain/packages/hrm-search-config/generateElasticsearchQuery.ts

```diff
@@ -274,6 +274,69 @@ const generateTranscriptQueriesFromFilters = ({
   }));
 };
 
+const generateContactNumberQueries = ({
+  searchParameters,
+  buildParams = { parentPath: '' },
+  queryWrapper = p => p,
+}: {
+  searchParameters: SearchParametersContact;
+  buildParams?: { parentPath: string };
+  queryWrapper?: (
+    p: DocumentTypeQueryParams[DocumentType],
+  ) => DocumentTypeQueryParams[DocumentType];
+}): QueryDslQueryContainer[] => {
+  const defaultQueries = [
+    generateQueryFromSearchTerms({
+      documentType: DocumentType.Contact,
+      fields: [
+        {
+          field: 'number',
+          boost: BOOST_FACTORS.contact,
+        },
+      ],
+      searchTerm: searchParameters.searchTerm,
+      parentPath: buildParams.parentPath,
+      queryWrapper,
+    }),
+  ];
+  const numberTerm = searchParameters.searchTerm.match(/\d+/g)?.join('');
+
+  if (
+    numberTerm &&
+    (searchParameters.searchTerm.startsWith('+') || numberTerm.length > 8)
```

Comment:
We probably don't want to limit number searches to those that start with a +, it would be good to support searches that include a phone number and other terms (i.e. if they have a large number of contacts from the same caller, they might want to narrow it further) and a lot of people don't prefix a plus sign when they type out a phone number